tests/conftest.py

```diff
@@ -36,20 +37,63 @@ def generate_random_file(filename, size):
         fout.write(os.urandom(size))
 
 
+def tmp_certificate(duration):
```

Comment:
Sometimes you call this both in server and client which causes some complexity.
I'd make this into several dependent fixtures:
* `ca` - first part of this function returning ca key, crt . with session scope, used in service
* `usercert(ca)` - using ca key, crt to produce valid user cert
* `alt_ca` - some other ca, not used in service at first